packages/@aws-cdk/aws-ecs/lib/base/pipeline-action.ts

```diff
@@ -0,0 +1,75 @@
+import codepipeline = require('@aws-cdk/aws-codepipeline-api');
+import { Repository } from '@aws-cdk/aws-ecr';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import cdk = require('@aws-cdk/cdk');
+import { BaseService } from './base-service';
+
+/**
+ * Common properties for the {@link PipelineSourceAction CodePipeline deploy Action},
+ * whether creating it directly,
+ * or through the {@link IRepository#toCodePipelineSourceAction} method.
+ */
+export interface PipelineDeployActionProps extends codepipeline.CommonActionProps {
+  /**
+   * the input artifact contains the image definitions file which describes the ECR docker image(s) to deploy
+   * @see https://docs.aws.amazon.com/codepipeline/latest/userguide/pipelines-create.html#pipelines-create-image-definitions
+   */
+  inputArtifact: codepipeline.Artifact;
+
+  /**
+   * the filename of the image definitions file in the input artifact
+   */
+  fileName: string
+
+  /**
+   * the service to deploy inside the ECS cluster
+   */
+  service: BaseService
+
+  /**
+   * the repository where the docker image(s) are stored
+   */
+  repository: Repository
+}
+
+/**
+ * CodePipelie Deploy Action to deploy the Service to ECS Cluster
+ */
+export class EcsDeployAction extends codepipeline.DeployAction {
```

Comment:
According to our current conventions, the name of this class should be `PipelineDeployAction`, so that it reads `ecs.PipelineDeployAction` when used.